### PR TITLE
refact: conda/mamba create/update env, fix user permissions

### PR DESCRIPTION
**Pull request by kemingy**
Signed-off-by: Keming <kemingyang@tensorchord.ai>

* split "create user envd" and "chown envd"
* fix conda/mamba create new env from env YAML file

### CODE REVIEW

File: pkg/lang/ir/conda.go [hunk 1]
```diff
@@ -66,97 +55,75 @@ func (g Graph) compileCondaChannel(root llb.State) llb.State {
...
+		env = fmt.Sprintf("--file %s ", g.CondaConfig.CondaEnvFileName) + env
+		// conda needs to use `conda env create` to create env from an env YAML file
+		if !g.CondaConfig.UseMicroMamba {
+			return fmt.Sprintf("%s env create %s", cmd, env)
+		}
+	}
+	return fmt.Sprintf("%s create %s", cmd, env)
+}
+
 func (g Graph) compileCondaPackages(root llb.State) llb.State {
```
Comment (gaocegege) [inline pkg/lang/ir/conda.go:86]:
```suggestion
func (g *Graph) compileCondaPackages(root llb.State) llb.State {
```

**Review by VoVAllen — COMMENTED** [review]
Do you need also chown on pip package path? Also if there's pip package in conda yaml file, is it installed under root user or envd user?

**Comment by VoVAllen** [comment]
Also will conda install anything under current user folder?

**Comment by kemingy** [comment]
> Do you need also chown on pip package path? Also if there's pip package in conda yaml file, is it installed under root user or envd user?

Pip package path should be under "/opt/conda/envs/envd".

Tested "install -e ." can work.

**Comment by kemingy** [comment]
> Also will conda install anything under current user folder?

Does "current user folder" mean "/home/envd"? This one has also been chowned.

**Comment by VoVAllen** [comment]
I mean if the installation process is executed under root user, will it install anything to `/root/XXXX`?

**Comment by gaocegege** [comment]
> Pip package path should be under "/opt/conda/envs/envd".

Could you please verify if pip cache works after the change?

**Comment by VoVAllen** [comment]
You can try install pip package such as numpy in conda. And check whether user can `import numpy` under user envd

**Comment by gaocegege** [comment]
And, if we use root in the building, and use envd in the runtime, there may be different cache locations between build time and runtime. 

For example, pip cache in build time will be stored in /home/root/.pip, and it should be /home/envd/.pip in runtime.

I am not sure if it is possible to customise the cache location.

**Comment by gaocegege** [comment]
Please rebase.

**Comment by gaocegege** [comment]
https://github.com/tensorchord/envd/actions/runs/3170574497/jobs/5163236483#step:7:3192

There are some issues in CI.

**Comment by kemingy** [comment]
According to my observation, `micromamba update -f env.yaml` doesn't install anything from the YAML file. Instead, it only updates all the packages.

We'd better use `conda update` because it requires mounts, which means it cannot be cached.

**Comment by kemingy** [comment]
I removed most of the `chown` operations. Users can still access `conda`, use `pip install`.

* fix #644

**Comment by kemingy** [comment]
`https://open-vsx.org/api` is too fragile. Sadly, GitHub actions don't support `allow-failure` https://github.com/actions/toolkit/issues/399.

**Comment by gaocegege** [comment]
/lgtm

Thanks for your contribution! :tada: :+1:

**Pull request merged**